sigma/cli/convert.py

```
@@ -286,14 +332,56 @@ def convert(
         elif correlation_method not in correlation_methods.keys():
             raise click.BadParameter(
                 f"Correlation method '{correlation_method}' is not supported by backend '{target}'. Run "
-                + click.style(f"sigma list correlation-methods {target}", bold=True, fg="green")
+                + click.style(
+                    f"sigma list correlation-methods {target}", bold=True, fg="green"
+                )
                 + " to list all available correlation methods of the target.",
                 param_hint="correlation_method",
             )
 
     try:
         rule_collection = load_rules(input + filter, file_pattern)
         result = backend.convert(rule_collection, format, correlation_method)
+        if output_dir:
+            writes_successful = True
+
+            # Collect all Paths for Rules
+            all_paths = []
+            for dir_path in input:
+                all_paths.extend(
+                    list(
+                        SigmaCollection.resolve_paths(
+                            [dir_path],
+                            recursion_pattern="**/" + file_pattern,
+                        )
+                    )
+                )
+            for index, path_of_input in enumerate(all_paths):
+                original_path_part_to_keep = os.path.sep.join(
+                    path_of_input.parts[-nesting_level:]
+                )
+
+                try:
+                    out_path = os.path.join(output_dir, original_path_part_to_keep)
+                    ensure_dir_exists(
+                        ctx=None, param=None, value=os.path.dirname(out_path)
+                    )
+                    with open(out_path, "w", encoding="utf-8") as f:
+                        f.write(result[index])
+                except Exception as ex:
+                    click.echo(
```

This should raise a `click.ClickException` instead of silently dropping the error state and just output a message.